src/notation/view/abstractnotationpaintview.cpp

```diff
@@ -497,7 +488,11 @@ void AbstractNotationPaintView::showContextMenu(const ElementType& elementType,
 void AbstractNotationPaintView::hideContextMenu()
 {
     TRACEFUNC;
-    emit hideContextMenuRequested();
+
+    if (m_isContextMenuOpen) {
+        emit hideContextMenuRequested();
+        m_isContextMenuOpen = false;
```

Comment:
I don't immediately see if this field will be set to `false` also when the context menu is closed because the user clicks outside it on some other part of the app or in another app. Are you sure that this happens properly? 

And do we really need to store this information in C++? Can't we emit the signal regardless of the state in QML, and do the check in QML itself (where we can directly access the popup itself)?

Reply:
Yes, you are right that there was a flaw in the previous implementation, but also, I don't think it's a good idea always emit this signal. I've changed the implementation: now we don't change these flags right after emitting the signals but change them when qml tells us that the popups are really open/closed